### PR TITLE
Added Docker support

### LINKED ISSUE #64: TZInfo::DataSourceNotFound - No source of timezone data could be found.
Looks like we are missing `tzinfo-data` when running in Ubuntu Focal in Docker

```
cd lib/app; bundle exec rackup --host 0.0.0.0 config.ru
[2022-03-31 10:54:34] INFO  WEBrick 1.4.4
[2022-03-31 10:54:34] INFO  ruby 2.6.9 (2021-11-24) [x86_64-linux]
[2022-03-31 10:54:34] INFO  WEBrick::HTTPServer#start: pid=32 port=9292
2022-03-31 10:54:39 - TZInfo::DataSourceNotFound - No source of timezone data could be found.
Please refer to https://tzinfo.github.io/datasourcenotfound for help resolving this error.:
	/app/vendor/bundle/ruby/2.6.0/gems/tzinfo-1.2.9/lib/tzinfo/data_source.rb:182:in `rescue in create_default_data_source'
	/app/vendor/bundle/ruby/2.6.0/gems/tzinfo-1.2.9/lib/tzinfo/data_source.rb:179:in `create_default_data_source'
	/app/vendor/bundle/ruby/2.6.0/gems/tzinfo-1.2.9/lib/tzinfo/data_source.rb:40:in `block in get'
	/app/vendor/bundle/ruby/2.6.0/gems/tzinfo-1.2.9/lib/tzinfo/data_source.rb:39:in `synchronize'
	/app/vendor/bundle/ruby/2.6.0/gems/tzinfo-1.2.9/lib/tzinfo/data_source.rb:39:in `get'
	/app/vendor/bundle/ruby/2.6.0/gems/tzinfo-1.2.9/lib/tzinfo/timezone.rb:665:in `data_source'
	/app/vendor/bundle/ruby/2.6.0/gems/tzinfo-1.2.9/lib/tzinfo/timezone.rb:92:in `get'
	/app/vendor/bundle/ruby/2.6.0/gems/tzinfo-1.2.9/lib/tzinfo/timezone.rb:113:in `new'
	/app/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.6/lib/active_support/values/time_zone.rb:206:in `find_tzinfo'
	/app/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.6/lib/active_support/values/time_zone.rb:300:in `initialize'
	/app/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.6/lib/active_support/values/time_zone.rb:234:in `new'
	/app/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.6/lib/active_support/values/time_zone.rb:234:in `[]'
	/app/vendor/bundle/ruby/2.6.0/gems/activesupport-5.2.6/lib/active_support/values/time_zone.rb:215:in `new'
	/app/lib/app/app.rb:57:in `block in <class:ConnectorsWebApp>'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1675:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1675:in `block in compile!'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1061:in `block in process_route'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1059:in `catch'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1059:in `process_route'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1000:in `block in filter!'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:999:in `each'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:999:in `filter!'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1126:in `block in dispatch!'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `block in invoke'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `catch'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `invoke'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1124:in `dispatch!'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:939:in `block in call!'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `block in invoke'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `catch'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1101:in `invoke'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:939:in `call!'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:929:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-protection-2.1.0/lib/rack/protection/xss_header.rb:18:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-protection-2.1.0/lib/rack/protection/path_traversal.rb:16:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-protection-2.1.0/lib/rack/protection/json_csrf.rb:26:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-protection-2.1.0/lib/rack/protection/base.rb:50:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-protection-2.1.0/lib/rack/protection/frame_options.rb:31:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/null_logger.rb:11:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:216:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1991:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1542:in `block in call'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1769:in `synchronize'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:1542:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/tempfile_reaper.rb:15:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/lint.rb:50:in `_call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/lint.rb:38:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/show_exceptions.rb:23:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/common_logger.rb:38:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/sinatra-2.1.0/lib/sinatra/base.rb:253:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/content_length.rb:17:in `call'
	/app/vendor/bundle/ruby/2.6.0/gems/rack-2.2.3/lib/rack/handler/webrick.rb:95:in `service'
	/root/.rbenv/versions/2.6.9/lib/ruby/2.6.0/webrick/httpserver.rb:140:in `service'
	/root/.rbenv/versions/2.6.9/lib/ruby/2.6.0/webrick/httpserver.rb:96:in `run'
	/root/.rbenv/versions/2.6.9/lib/ruby/2.6.0/webrick/server.rb:307:in `block in start_thread'
```

### PR DESCRIPTION
**Pull request by tarekziade**
Adds a Dockerfile that installs and deploys the app using Ubuntu Focal

Also Fixes #64

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,15 +25,21 @@ build:
...
 	${YQ} e ".revision = \"$(shell git rev-parse HEAD)\"" -i config/connectors.yml
 	${YQ} e ".repository = \"$(shell git config --get remote.origin.url)\"" -i config/connectors.yml
 	${YQ} e ".version = \"$(shell script/version.sh)\"" -i config/connectors.yml
-	cd lib/app; bundle exec rackup config.ru
+	cd lib/app; bundle exec rackup --host 0.0.0.0 config.ru
+
+build-docker:
+	docker build -t connectors .
+
+run-docker:
```
Comment (artem-shelkovnikov) [inline Makefile:41]: Does it depend on `build-docker`? If so, should we also declare the dependency and build the image if it's not built?

**Review by artem-shelkovnikov — APPROVED** [review]
LGTM, added a small question

File: Makefile (hunk 1 above)
Comment (tarekziade) [reply Makefile:41]: yes and no. if you run it without the image built, it'll fail. But it should run without re-buidling everytime. I can't think of a simple mechanism to trigger the build if it was not done when running

Comment (artem-shelkovnikov) [reply Makefile:41]: Fair point, I was also on the fence whether it should be a hard requirement. So if the docker image was never built, we still get a somewhat meaningful error message, right?

Comment (tarekziade) [reply Makefile:41]:
```
% docker run --rm -it -p 127.0.0.1:9292:9292/tcp connectors
Unable to find image 'connectors:latest' locally
docker: Error response from daemon: pull access denied for connectors, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
```

**Review by seanstory — COMMENTED** [review]
I'm late to the party, but small comment.

File: Makefile [hunk 2]
```diff
@@ -25,15 +25,21 @@ build:
...
+	docker run --rm -it -p 127.0.0.1:9292:9292/tcp connectors
```
[the remark this answers is not in the document]
Comment (tarekziade) [reply Makefile:42]: Yeah you're right, we should not hardcore the port. For the env it sounds like a great improvement

[the remark this answers is not in the document]
Comment (seanstory) [reply Makefile:42]: Filed an issue for this: https://github.com/elastic/enterprise-search-team/issues/1765

**Pull request merged**